account/account.go

```diff
@@ -387,42 +416,19 @@ func dataAvailabilityMode(feeDAMode, nonceDAMode rpc.DataAvailabilityMode) (uint
 // - error: an error, if any
 //
 // If the `tx` parameter is not one of the supported types, the function returns an error `ErrTxnTypeUnSupported`.
-func (account *Account) TransactionHashDeclare(tx rpc.DeclareTxnType) (*felt.Felt, error) {
+func (account *Account) TransactionHashDeclare(tx rpc.BroadcastDeclareTxnType) (*felt.Felt, error) {
 
 	switch txn := tx.(type) {
-	case rpc.DeclareTxnV0:
-		// Due to inconsistencies in version 0 hash calculation we don't calculate the hash
+	case rpc.BroadcastDeclareTxnV1:
+		// Due to inconsistencies in version 1 hash calculation we don't calculate the hash
 		return nil, ErrTxnVersionUnSupported
-	case rpc.DeclareTxnV1:
-		if txn.SenderAddress == nil || txn.Version == "" || txn.ClassHash == nil || txn.MaxFee == nil || txn.Nonce == nil {
-			return nil, ErrNotAllParametersSet
-		}
-
-		calldataHash, err := hash.ComputeHashOnElementsFelt([]*felt.Felt{txn.ClassHash})
-		if err != nil {
-			return nil, err
-		}
 
-		txnVersionFelt, err := new(felt.Felt).SetString(string(txn.Version))
-		if err != nil {
-			return nil, err
-		}
-		return hash.CalculateTransactionHashCommon(
-			PREFIX_DECLARE,
-			txnVersionFelt,
-			txn.SenderAddress,
-			&felt.Zero,
-			calldataHash,
-			txn.MaxFee,
-			account.ChainId,
-			[]*felt.Felt{txn.Nonce},
-		)
-	case rpc.DeclareTxnV2:
-		if txn.CompiledClassHash == nil || txn.SenderAddress == nil || txn.Version == "" || txn.ClassHash == nil || txn.MaxFee == nil || txn.Nonce == nil {
+	case rpc.BroadcastDeclareTxnV2:
+		if txn.CompiledClassHash == nil || txn.SenderAddress == nil || txn.Version == "" || txn.MaxFee == nil || txn.Nonce == nil {
```

Comment:
Let's validate if the ContractClass is present with something like this: `txn.ContractClass.SierraProgram == nil`